contracts/TokenFactory.sol

```diff
@@ -335,28 +321,63 @@ contract TokenFactory is ReentrancyGuard, Ownable {
 
         address winnerToken = getWinnerByCompetitionId(_competitionId);
 
-        require(winnerToken != tokenAddress, "token address is the winner");
-
         Token token = Token(tokenAddress);
-        uint256 burnedAmount = token.balanceOf(msg.sender);
-
-        uint256 receivedETH = _sell(
-            tokenAddress,
-            burnedAmount,
-            msg.sender,
-            address(this)
-        );
 
-        uint256 mintedAmount = _buy(winnerToken, msg.sender, receivedETH);
-
-        emit BurnTokenAndMintWinner(
-            msg.sender,
-            tokenAddress,
-            winnerToken,
-            burnedAmount,
-            receivedETH,
-            mintedAmount,
-            block.timestamp
-        );
+        if(winnerToken == tokenAddress) {
```

Comment:
Use a different function for "graduating to Uniswap"? Since the logic is completely different from burning losing tokens